llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -1545,6 +1545,53 @@ Instruction *InstCombinerImpl::foldICmpTruncConstant(ICmpInst &Cmp,
   return nullptr;
 }
 
+/// Fold icmp (trunc X), (trunc Y).
+/// Fold icmp (trunc X), (zext Y).
+static Instruction *foldICmpTruncWithTruncOrExt(ICmpInst &Cmp,
+                                                InstCombinerImpl &IC,
+                                                const SimplifyQuery &Q) {
+  if (!Cmp.isEquality() && !Cmp.isUnsigned())
```

Comment:
```suggestion
  if (Cmp.isSigned())
```

Reply:
Agreed less code although think the original shows intents of the lemma better. But no strong feelings. Changing.